tests/jballerina-unit-test/src/test/resources/test-src/main.function/operand/invalid_different_array.bal

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public function main(int[] a, string... d) {
```

Comment:
Are these invalid cases are covered by tests.
main(int[] a, decimal[] b)
main(int[] a, int[] b)
main(int [], int a= 10)
main(int[], int? b)
main(int? a)

And following valid cases
main(string[] a, string b)
main(int? a, string b)

Reply:
Yes these cases are covered. I think `main(int? a)` is a valid scenario. So it is not covered as an error.
Also `main(int? a, string b)` is an error so it is covered as an error
Note that the exact same types given here are not used in tests, but different types are used to validate the required error.

Reply:
As per all operand parameters of type O? must follow any operand parameter of type O
where O is a type that is a subtype of one of the supported basic types

Reply:
is this main(int? a) valid?

Reply:
I thought because of the word `any`, if there is no operand parameter of type `O` then type `O?` need not follow operand parameter of type `O`, it can remain without it